influence_benchmark/environment/environment.py

```diff
@@ -1,61 +1,133 @@
 import random
-from typing import Dict, Tuple, Any, List
-import openai
+
+import yaml
+
+from influence_benchmark.environment.character import character
+from influence_benchmark.environment.state import State
+from influence_benchmark.environment.transition_model import TransitionModel
 
 
 class Environment:
-    def __init__(self, config, env_name: str):
+    def __init__(self, config: dict):
         self.config = config
-        self.state = None
-        self.env_name = env_name
-
-    def reset(self):
-        self.state = self.generate_initial_state()
+        self.env_name = config["env_name"]
+        self.backend = config["env_backend_model"]
+        self.transition_model = TransitionModel(self.env_name, "gpt-4o")
+        with open(f"influence_benchmark/config/env_configs/{self.env_name}.yaml", "r") as file:
+            data = yaml.safe_load(file)
+        self.variables = {}
+        possible_vars = data["possible_env_vars"]
+
+        if "possible_env_vars" in data:
+            for key in possible_vars:
+                self.variables[key] = random.choice(possible_vars[key])
```

Comment:
Probably worth extracting this as its own function for readability, e.g.

`read_config_and_set_variables`